include/grpcpp/security/tls_credentials_options.h

```diff
@@ -0,0 +1,260 @@
+  }
+
+  void Cancel(TlsCredentialReloadArg* arg) const {
+    if (cancel_ == nullptr) {
+      gpr_log(GPR_ERROR, "cancel API is nullptr");
+      return;
+    }
+    cancel_(config_user_data_, arg);
+  }
+  /** Returns a C struct for the credential reload config. **/
+  grpc_tls_credential_reload_config* c_credential_reload() const {
+    return c_config_;
+  }
+
+ private:
+  grpc_tls_credential_reload_config* c_config_;
+  void* config_user_data_;
+  int (*schedule_)(void* config_user_data, TlsCredentialReloadArg* arg);
+  void (*cancel_)(void* config_user_data, TlsCredentialReloadArg* arg);
+  void (*destruct_)(void* config_user_data);
+};
+
+/** TLS server authorization check arguments, wraps
+ *  grpc_tls_server_authorization_check_arg. **/
+
+class TlsServerAuthorizationCheckArg {
+ public:
+  TlsServerAuthorizationCheckArg();
+  TlsServerAuthorizationCheckArg(grpc_tls_server_authorization_check_arg arg);
+  ~TlsServerAuthorizationCheckArg();
+
+  /** Getters for member fields. **/
+  void* cb_user_data() const;
+  int success() const;
+  std::shared_ptr<grpc::string> target_name() const;
+  std::shared_ptr<grpc::string> peer_cert() const;
+  grpc_status_code status() const;
+  std::shared_ptr<grpc::string> error_details() const;
+
+  /** Setters for member fields. **/
+  void set_cb_user_data(void* cb_user_data);
+  void set_success(int success);
+  void set_target_name(const grpc::string& target_name);
+  void set_peer_cert(const grpc::string& peer_cert);
+  void set_status(grpc_status_code status);
+  void set_error_details(const grpc::string& error_details);
+
+  /** Calls the C arg's callback function. **/
+  void callback();
+
+ private:
+  grpc_tls_server_authorization_check_arg c_arg_;
+};
+
+// Exposed for testing purposes.
+int tls_server_authorization_check_config_c_schedule(
+    void* config_user_data, grpc_tls_server_authorization_check_arg* arg);
+void tls_server_authorization_check_config_c_cancel(
+    void* config_user_data, grpc_tls_server_authorization_check_arg* arg);
+
+/** TLS server authorization check config, wraps
+ *  grps_tls_server_authorization_check_config. **/
+class TlsServerAuthorizationCheckConfig {
+ public:
+  TlsServerAuthorizationCheckConfig(
+      const void* config_user_data,
+      int (*schedule)(void* config_user_data,
+                      TlsServerAuthorizationCheckArg* arg),
+      void (*cancel)(void* config_user_data,
+                     TlsServerAuthorizationCheckArg* arg),
+      void (*destruct)(void* config_user_data));
+  ~TlsServerAuthorizationCheckConfig();
+
+  int Schedule(TlsServerAuthorizationCheckArg* arg) const {
+    return schedule_(config_user_data_, arg);
+  }
+
+  void Cancel(TlsServerAuthorizationCheckArg* arg) const {
+    if (cancel_ == nullptr) {
+      gpr_log(GPR_ERROR, "cancel API is nullptr");
+      return;
+    }
+    cancel_(config_user_data_, arg);
+  }
+
+  /** Creates C struct for the credential reload config. **/
```

Comment:
Please also update the comment accordingly. It is not a credential reload config :)